### PR TITLE
Fix for issue #116

### PR DESCRIPTION
**Pull request by marcosdejesus**
This PR address issue #116

 This bug fix insert some checks to build the SAFE structure of reprocessed Sentinel-2 L2A products appropriately.

### CODE REVIEW

File: sentinelhub/aws_safe.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def get_safe_struct(self):
         if self.data_collection is DataCollection.SENTINEL2_L2A:
             for mask in AwsConstants.CLASS_MASKS:
                 for resolution in [AwsConstants.R20m, AwsConstants.R60m]:
-                    if self.baseline <= '02.06':
+                    if self.baseline <= '02.06' or self.baseline == '00.01':
```
Comment (AleksMat) [inline sentinelhub/aws_safe.py:240]: It seems that everywhere else baseline `00.01` is handled the same as baselines `02.07` and above. Only here baseline `00.01` is handled the same as baselines `02.06` and below. Is there a reason for this?

Note that this results into a different naming of 4 metadata masks.

- this is what we get at the moment:
```
T33UUQ_20170414T101021_SNW_20m.jp2
T33UUQ_20170414T101021_SNW_60m.jp2
T33UUQ_20170414T101021_CLD_20m.jp2
T33UUQ_20170414T101021_CLD_60m.jp2
```
- this is what we could get for baselines `>=02.07`
```
MSK_SNWPRB_20m.jp2
MSK_SNWPRB_60m.jp2
MSK_CLDPRB_20m.jp2
MSK_CLDPRB_60m.jp2
```
- and this is what we could get for baselines `<=02.06` (if everything would be consistent with those):
```
L2A_T33UUQ_20170414T101021_SNW_20m.jp2
L2A_T33UUQ_20170414T101021_SNW_60m.jp2
L2A_T33UUQ_20170414T101021_CLD_20m.jp2
L2A_T33UUQ_20170414T101021_CLD_60m.jp2
```

So the current naming of these masks for `00.01` is not consistent neither with the old baselines nor with the newer.

**Review by AleksMat — COMMENTED** [review]
Looks great. I only have one comment/question that I would like to discuss before we merge.

File: sentinelhub/aws_safe.py (hunk 1 above)
Comment (marcosdejesus) [reply sentinelhub/aws_safe.py]: You are completely right.

I just changed this part to treat the reprocessed files as baselines `02.07` and above.

Comment (AleksMat) [reply sentinelhub/aws_safe.py]: Excellent, all clear now.

**Comment by AleksMat** [comment]
Thanks for the contribution. It will be included in today's release. :+1:

**Pull request merged**